src/graph/Sankey.js

```diff
@@ -0,0 +1,232 @@
+            .attr("d", this._d3SankeyPath)
+            .style("stroke-width", function (d) { return Math.max(1, d.dy); })
+            .sort(function (a, b) { return b.dy - a.dy; })
+        ;
+        link.select("title")
+            .text(function (d) {
+                return d.source.name + " → " + d.target.name + "\n" + d.value;
+            })
+        ;
+        link.exit().remove();
+
+        // Nodes ---
+        var node = element.selectAll(".node").data(sankeyData.vertices);
+        node.enter().append("g")
+            .attr("class", "node")
+            .each(function (d) {
+                var gElement = d3.select(this);
+                gElement.append("rect");
+                gElement.append("text");
+            })
+            .on("click.mouse", function (d, idx) {
+                context.click(d.origRow, "", true);
+            })
+            .call(d3.behavior.drag()
+                .origin(function (d) { return d; })
+                .on("dragstart", function () {
+                    this.parentNode.appendChild(this);
+                })
+                .on("drag", dragmove)
+            )
+        ;
+        node
+            .attr("transform", function (d) {
+                return "translate(" + d.x + "," + d.y + ")";
+            })
+        ;
+        node.select("rect")
+            .attr("height", function (d) { return d.dy; })
+            .attr("width", this._d3Sankey.nodeWidth())
+            .style("fill", function (d) {
+                return context._palette(d.name);
+            })
+            .style("stroke", function (d) {
+                return d3.rgb(context._palette(d.name)).darker(2);
+            })
+            .style("cursor", (this.xAxisMovement() || this.yAxisMovement()) ? null : "default")
+        ;
+        node.select("text")
+             .attr("x", -6)
+             .attr("y", function (d) { return d.dy / 2; })
+             .attr("dy", ".35em")
+             .attr("text-anchor", "end")
+             .attr("transform", null)
+             .text(function (d) { return d.name; })
+           .filter(function (d) { return d.x < context.width() / 2; })
+             .attr("x", 6 + this._d3Sankey.nodeWidth())
+             .attr("text-anchor", "start")
+        ;
+        node.exit().remove();
+
+        function dragmove(d) {
+            var gElement = d3.select(this);
+            if (context.xAxisMovement()) {
+                d.x = Math.max(0, Math.min(context.width() - d.dx, d3.event.x));
+            }
+            if (context.yAxisMovement()) {
+                d.y = Math.max(0, Math.min(context.height() - d.dy, d3.event.y));
+            }
+            gElement.attr("transform", "translate(" + d.x + "," + d.y + ")");
+            context._d3Sankey.relayout();
+            link.attr("d", context._d3SankeyPath);
+
+            gElement.select("text")
+                .attr("x", -6)
+                .attr("y", function (d) { return d.dy / 2; })
+                .attr("dy", ".35em")
```

Comment:
This is ok - its an approximation to vertical align text (as the baseline is about a thirs of the way up from the bottom).  And 1 em is 1 Char Height.